### PR TITLE
Use ValueTasks

### PR DESCRIPTION
**Pull request by iamcarbon**
This PR replaces the Task with ValueTask.

### CODE REVIEW

**Comment by iamcarbon** [comment]
@bleroy Ready for review.

**Review by bleroy — CHANGES_REQUESTED** [review]
Looks fine, thanks for doing this. My only feedback is to use the `ValueTask.FromResult` static where the original code was using `Task.FromResult`

File: LunrCore/AsyncEnumerableExtensions.cs [hunk 1]
```diff
@@ -41,10 +41,10 @@ await foreach (TSource sourceItem in source.WithCancellation(cancellationToken))
             this IEnumerable<T> source,
             [EnumeratorCancellation] CancellationToken cancellationToken)
         {
-            foreach(T item in source)
+            foreach (T item in source)
             {
                 if (cancellationToken.IsCancellationRequested) yield break;
-                yield return await Task.FromResult(item).ConfigureAwait(false);
+                yield return await new ValueTask<T>(item).ConfigureAwait(false);
```
[the remark this answers is not in the document]
Comment (iamcarbon) [reply LunrCore/AsyncEnumerableExtensions.cs:47]: @bleroy ValueTask.FromResult is not available on netstandard2.0.

[the remark this answers is not in the document]
Comment (bleroy) [reply LunrCore/AsyncEnumerableExtensions.cs:47]: Makes sense.

**Comment by bleroy** [comment]
Thanks for the contribution.

**Pull request merged**